lib/IRGen/GenStruct.cpp

```diff
@@ -165,36 +165,31 @@ namespace {
     }
 
     /// Given a full struct explosion, project out a single field.
-    void projectFieldFromExplosion(IRGenFunction &IGF,
-                                   Explosion &in,
-                                   VarDecl *field,
-                                   Explosion &out) const {
+    virtual void projectFieldFromExplosion(IRGenFunction &IGF, Explosion &in,
+                                           VarDecl *field,
+                                           Explosion &out) const {
```

Comment:
Why has this been made virtual?

Reply:
If we have a base class, the offset changes. So we need to project starting at the first member offset, not at `0`.